controllers/openstackdataplane_controller.go

```diff
@@ -205,6 +213,12 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 			err.Error()))
 		return ctrl.Result{}, err
 	}
+	if !foundRoles {
+		r.Log.Info("couldn't find roles with the label", "openstackdataplane", instance.Name)
+		instance.Status.Conditions.Set(condition.FalseCondition(condition.DeploymentReadyCondition, condition.RequestedReason, condition.SeverityInfo, condition.DeploymentReadyInitMessage))
```

Comment:
This would be the reason you had to change the dataplane-create-test. You set the DeploymentReadyCondition to Requested, even though a deployment has not  been requested.

Reply:
I don't think you need to set/change any status condition here at all. Why not just move this entire "if" to before the "if" for the deploy. And then not mark SetupReady as True until all the roles are ready.

Reply:
done!